pageserver/src/tenant/timeline.rs

```diff
@@ -5358,6 +5358,90 @@ impl Timeline {
             shard_count: self.tenant_shard_id.shard_count,
         }
     }
+
+    #[cfg(test)]
+    pub(super) fn force_advance_lsn(self: &Arc<Timeline>, new_lsn: Lsn) {
+        self.last_record_lsn.advance(new_lsn);
+    }
```

Comment:
Well... Since this is always advancing, perhaps there is nothing bad happening. I was thinking if we need a warning label on the method, but probably not.